### PR TITLE
Fixes for stm32

### PR DESCRIPTION
**Pull request by ricardosalveti**
Minor cleanups for stm32, and fixing the base build for eval.

Eval is not yet fully functional due flashlayouts, which should be changed for an image type as currently it is hardcoding the image version which doesn't work correctly based on the image build. It is also looking for a missing fip bin, but didn't yet check why.

### CODE REVIEW

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc [hunk 1]
```diff
@@ -414,30 +414,25 @@ BOOTSCHEME_LABELS:stm32mp1common += "optee"
...
 LINUX_A7_EXAMPLES_DT:remove:stm32mp15-eval = "stm32mp157c-ev1-a7-examples"
 CUBE_M4_EXAMPLES_DT:remove:stm32mp15-eval = "stm32mp157c-ev1-m4-examples"
 UBOOT_DEVICETREE:stm32mp15-eval = "stm32mp157c-ev1"
 IMAGE_FSTYPES:remove:stm32mp15-eval = "wic wic.gz wic.bmap"
 EXTRA_IMAGEDEPENDS:append:stm32mp15-eval = "flashlayouts-stm32mp1"
-UBOOT_LOADADDRESS:stm32mp15-eval = "0xC2000000"
-UBOOT_ENTRYPOINT:stm32mp15-eval = "0xC2000000"
-UBOOT_DTB_LOADADDRESS:stm32mp15-eval = "0xC4000000"
-UBOOT_RD_LOADADDRESS:stm32mp15-eval = "0xC4400000"
-UBOOT_RD_ENTRYPOINT:stm32mp15-eval = "0xC4400000"
```
Comment (angolini) [inline meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:427]: I don't see UBOOT_RD* variables for the target common. They are only being removed. I will assume this is on purpose

**Review by angolini — APPROVED** [review]
LGTM

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:427]: Yes, they are not needed.

**Review by Tim-Anderson — APPROVED** [review]
LGTM

**Comment by ricardosalveti** [comment]
@igoropaniuk let me know if you want me to merge this as is for adding your extra flashlayout changes on top.

**Comment by igoropaniuk** [comment]
LGTM

**Pull request merged**